app/lib/frontend/templates/views/pkg/install_tab.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../../../../package/models.dart';
+import '../../../../shared/tags.dart';
+import '../../../dom/dom.dart' as d;
+
+d.Node installTabNode({
+  required PackageVersion version,
+  required List<String>? tags,
+  required bool isDevDependency,
+}) {
+  final packageName = version.package;
+  List<_ImportExample> importExamples;
+  if (version.libraries!.contains('$packageName.dart')) {
+    importExamples = <_ImportExample>[
+      _ImportExample(packageName, '$packageName.dart'),
+    ];
+  } else {
+    importExamples = version.libraries!
+        .map((library) => _ImportExample(packageName, library))
+        .toList();
+  }
+
+  final executables = version.pubspec!.executables?.keys.toList();
+  executables?.sort();
+  final hasExecutables = executables != null && executables.isNotEmpty;
+  final useAsLibrary = !hasExecutables || importExamples.isNotEmpty;
+
+  return d.fragment([
+    if (hasExecutables) _useAsExecutable(version.package, executables!),
+    if (useAsLibrary) _useAsLibrary(version, tags, isDevDependency),
+    if (importExamples.isNotEmpty) _useAsImport(importExamples),
+  ]);
+}
+
+d.Node _useAsExecutable(String package, List<String> executables) {
+  return d.fragment([
+    d.h2(text: 'Use this package as an executable'),
+    d.h3(text: 'Install it'),
+    d.p(text: 'You can install the package from the command line:'),
+    d.pre(
+      attributes: {'data-text-to-copy': 'dart pub global activate $package'},
+      child: d.code(
+        classes: ['language-shell'],
+        child: d.strong(text: 'dart pub global activate $package'),
+      ),
+    ),
```

Comment:
Why don't we make a helper method for `codeSnippet(language: ..., text: , child: , children: )`.

Then we don't have to write `pre` and `code`, and all the attributes.. ideally, we don't even need to supply `data-text-to-copy` if `text` is given..